webapp/src/app.tsx

```diff
@@ -3512,7 +3515,7 @@ export class ProjectView
             .finally(() => core.hideLoading("tutorial"));
     }
 
-    startActivity(activity: pxt.editor.Activity, path: string, title?: string, editorProjectName?: string) {
+    startActivity(activity: pxt.editor.Activity, path: string, title?: string, editorProjectName?: string, focus?: boolean) {
```

Comment:
```suggestion
    startActivity(activity: pxt.editor.Activity, path: string, title?: string, editorProjectName?: string, focus = true) {
```

and you can change portion below to

```typescript
this.textEditor.giveFocusOnLoading = focus;
```

Reply:
don't the optional parameters need to be last though? I can move everything XD I guess I was just scared of changing all the calls to startActivity

Reply:
oh lol I just tried it and it works. Dunno why I thought the optional parameters needed to be last!

Reply:
Optional parameters have to be set after required parameters, but optional parameters and default parameters can be mixed / order doesn't matter -- default parameters just emit to javascript with an extra if at the front of the function e.g.

![image](https://user-images.githubusercontent.com/5615930/91915901-99db4d80-ec70-11ea-934f-0d100fbcba37.png)

Reply:
oooooo cool thanks Joey!